src/MPoly.jl

```diff
@@ -445,6 +448,13 @@ function content(a::MPolyRingElem{T}) where T <: RingElement
    return z
 end
 
+
+function is_nilpotent(f::T) where {T<:MPolyRingElem}
+  is_domain_type(elem_type(coefficient_ring(f))) && return is_zero(f)
+  return all(is_nilpotent, coefficients(f))
+end
+
+
 function is_zero_divisor(x::MPolyRingElem{T}) where T <: RingElement
```

Comment:
Unrelated to this PR, but I wonder: should we also optimize this for domain types
```suggestion
function is_zero_divisor(x::MPolyRingElem{T}) where T <: RingElement
  is_domain_type(T) && return is_zero(f)
```

Reply:
Seems reasonable.  I'll try it -- just to make sure there are no unintended consequences.

Reply:
It works fine; not sure it brings any measurable benefit, but the code is there now.